src/components/SubscriptionListItem.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+	<ListItem :title="getTitle()"
+		:details="getDetails()">
+		<template #icon>
+			<Avatar :size="44"
+				:url="getImageSrc()"
+				:display-name="getAvatarName()" />
+		</template>
+		<template #subtitle>
+			<span v-if="isLoading"><em>(Loading RSS data...)</em></span>
+			<span v-else>{{ getSubtitle() }}</span>
+		</template>
+		<template #actions>
+			<ActionLink :href="getHomepageLink()"
+				target="_blank"
+				icon="icon-external">
+				Podcast's homepage
+			</ActionLink>
+			<ActionLink :href="getRssLink()"
+				target="_blank">
+				<template #icon>
+					<Rss />
+				</template>
+				RSS feed
+			</ActionLink>
+		</template>
+	</ListItem>
+</template>
+
+<script>
+import ActionLink from '@nextcloud/vue/dist/Components/ActionLink'
+import Avatar from '@nextcloud/vue/dist/Components/Avatar'
+import ListItem from '@nextcloud/vue/dist/Components/ListItem'
+
+import Rss from 'vue-material-design-icons/Rss.vue'
+
+import { generateUrl } from '@nextcloud/router'
+import axios from '@nextcloud/axios'
+
+export default {
+	name: 'SubscriptionListItem',
+	components: {
+		ActionLink,
+		Avatar,
+		ListItem,
+		Rss,
+	},
+	props: {
+		sub: {
+			type: Object,
+			required: true,
+		},
+	},
+	data() {
+		return {
+			podcastData: null,
+			isLoading: true,
+		}
+	},
+	async mounted() {
+		try {
+			const resp = await axios.get(generateUrl('/apps/gpoddersync/personal_settings/podcast_data?url={url}', {
+				url: this.sub.url,
+			}))
+			this.podcastData = resp.data?.data
+		} catch (e) {
+			console.error(e)
+		} finally {
+			this.isLoading = false
+		}
+	},
+	methods: {
+		getTitle() {
+			return this.podcastData?.title ?? this.sub.url ?? ''
+		},
+		getDetails() {
```

Comment:
please extract the calculation to another js file as to make it testable

Reply:
You need help with that?

Reply:
Yes please :)